File: cpp/sum-of-digits-of-string-after-convert.cpp
#include <string>

#include "iostream"


using namespace std;

class Solution {
public:
  int getLucky(string s, int k) {
    string num = "";

    for (char c : s) {
      int n = c - 'a' + 1;
      num.append(to_string(n));
    }

    for (int i = 0; i < k; i++) {
      int n = 0;

      for (char c : num) {
        n += c - '0' ;
      }

      num = to_string(n);
    }

    return stoi(num);
  }
};

int main() {
  Solution s;

  cout << s.getLucky("zzzj", 2);

  return 0;
}

File: java/sum-of-digits-of-string-after-convert.java


class Solution {
    public int getLucky(String s, int k) {

        StringBuilder sb = new StringBuilder();
        for (char c : s.toCharArray()) {
            sb.append(c - 'a' + 1);
        }

        String str = sb.toString();
        int sum = 0;
        for (int i = 0; i < k; i++) {
            sum = 0;
            for (char c : str.toCharArray()) {
                sum += c - '0';
            }
            str = String.valueOf(sum);
        }

        return sum;


    }
}